torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -546,6 +549,8 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         # All-gather full parameters, moving them to compute_device if
         # necessary.
         self._rebuild_full_params()
+        # Wait for all_gather full parameters to finish before computation
+        torch.cuda.current_stream().wait_stream(self._streams["all_gather"])
```

Comment:
Is this needed due to other changes in this PR, or is it a correctness fix?

Reply:
yes, it is needed, because I removed all_gather stream sync from _rebuild_full_params(), so we need to add it here.  all_gather stream sync is removed from _rebuild_full_params() because all gather does not need to be waited by current computation stream during prefetching call